WordPress/gutenberg
# Extract header used in PostVisibility and PublishDateTimePicker to a new InspectorPopoverHeader component

## What?
Adds a new `InspectorPopoverHeader` component which is used by `PostVisibility`, `PublishDateTimePicker`, and [more to come](https://github.com/WordPress/gutenberg/issues/39082).

## Why?
The headers in `PostVisibility`, `PublishDateTimePicker`, and the popovers in the [new sidebar design](https://github.com/WordPress/gutenberg/issues/39082) are identical so we should not duplicate the code.

## How?
The new component is in `@wordpress/block-editor` so that `PublishDateTimePicker`, which is used by the Post Date block, can use it. Another option is `@wordpress/components` (and name it `PopoverHeader`) but I'm not 100% convinced that this is a pattern we'll see in other parts of WordPress. It's easier to move from `@wordpress/block-editor` → `@wordpress/components` than the other way around so I think best to start off here.

I opted to go with `InspectorPopoverHeader` instead of `InspectorPopover` because that fits in nicely with how we've organised the existing popovers which is that the content of the popover is a seperate component passed into `Dropdown`'s `renderContent`.

Naming things is hard. I went with `InspectorPopoverHeader` which is a fairly literal description of its intended use. Open to suggestions.

## Testing Instructions
1. Create a post.
2. Open the post settings sidebar.
3. Change the _Publish_ date.
4. Change the _Visibility_.
5. Both popovers should look and feel good.

## Screenshots or screencast 
<img width="279" alt="Screen Shot 2022-05-30 at 11 20 18" src="https://user-images.githubusercontent.com/612155/170901473-2b5dd5d5-f8ea-4c1c-8074-0891d4fde49b.png">
<img width="278" alt="Screen Shot 2022-05-30 at 11 20 27" src="https://user-images.githubusercontent.com/612155/170901477-7bee7faa-73ac-4adb-8bab-0bc6d7a6bf42.png">
